src/groth16/prover.rs

```diff
@@ -313,18 +313,18 @@ where
         return Err(SynthesisError::UnexpectedIdentity);
     }
 
-    let mut g_a = vk.delta_g1 * &r;
+    let mut g_a = vk.delta_g1 * r;
     AddAssign::<&E::G1Affine>::add_assign(&mut g_a, &vk.alpha_g1);
-    let mut g_b = vk.delta_g2 * &s;
+    let mut g_b = vk.delta_g2 * s;
     AddAssign::<&E::G2Affine>::add_assign(&mut g_b, &vk.beta_g2);
     let mut g_c;
     {
         let mut rs = r;
         rs.mul_assign(&s);
 
-        g_c = vk.delta_g1 * &rs;
-        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.alpha_g1 * &s));
-        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.beta_g1 * &r));
+        g_c = vk.delta_g1 * rs;
+        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.alpha_g1 * s));
```

Comment:
These changes break compilation, so we can't address the clippy lint in this way.